modules/porous_flow/doc/content/modules/porous_flow/boundaries.md

```diff
@@ -94,50 +97,62 @@ environment.  The flux of fluid from the model to this environment is
+a value that ends up multiplying $C$ to give a flux. $C$ can be thought of as the conductance and is specified with `flux_function = C`. $P_{\mathrm{e}}$ can be specified using an AuxVariable or set to a constant value using `PT_shift = Pe`.
+
+
+!media piecewiselinear_g_function.png style=width:50%;margin-left:10px caption=Depiction of $g(P-P_{\mathrm{e}})$ for PorousFlowPiecewiseLinearSink. The function accepts $P-P_{\mathrm{e}}$ as an input (i.e. the difference between a specified environment pressure and the pressure on the boundary element) and returns a value that multiplies $C$ to give the flux out of the domain. id=PiecewiseLinear_g_Function
+
+To set a Dirichlet boundary condition $P = P_{\mathrm{e}}$, either $C$ or the slope of $g$ should be very large. 
+It is usually convenient to make the slope of $g$ equal to 1 by setting `pt_vals = '-1E9 1E9'` and `multipliers = '-1E9 1E9'`, and then $C$ can be selected appropriately.
+Assigning a large range for `pt_vals` ensures that $g$ is defined for the porepressures encountered in the simulation, and defining $g$ for $P-P_{\mathrm{e}}>0$ and $P-P_{\mathrm{e}}<0$ allows for outflow and inflow.
+If $P - P_{\mathrm{e}}$ falls outside of the range defined in `pt_vals`, then $g = 0$ by default. This can be useful to set a boundary condition that will only allow for outflow (e.g. by using  `pt_vals = '0 1E9'`, `multipliers = '0 1E9'`).`
+
+
+The numerical value of the conductance, $C$, is
 $\rho k_{nn}k_{\mathrm{r}}/\mu/L$, must be set at an appropriate value
 for the model.
-
 Alternately, a
 [`PorousFlowPiecewiseLinearSink`](PorousFlowPiecewiseLinearSink.md)
 may be constructed with the same `pt_vals`, `multipliers` and `PT_shift`, but with `use_mobility = true` and `use_relperm =
 true`, and then $C = 1/L$.  This has three advantages: (1) the MOOSE
 input file is simpler; (2) MOOSE automatically chooses the correct
 mobility and relative permeability to use; (3) these parameters are
 appropriate to the model so it reduces the potential for difficult
-numerical situations occuring.
+numerical situations occuring. Also note, if $C \times g$ is too large, the boundary residual will be much larger than residuals within the domain. This results in poor convergance. 
+
+So what value should be assigned to $C$? In the example below, $C = 10^{-6}$, $\rho \sim 10^3$ kg/m$^3$, $k = 10^{-15}$ m$^2$, $k_r = 1$, and $\mu \sim 10^{-3}$ Pa-s. Therefore $L \sim 10^{-3}$ m. This value of $L$ is small enough to ensure that the Dirichlet boundary condition is satisfied. If $C$ is increased to $10^{-2}$, $L \sim 10^{-7}$ m, and the simulation has difficulty converging. If $C = 10^{-11}$, $L\sim10^2$ m, and the boundary acts like a source of fluid from a distant reservoir (i.e. it no longer acts like a Dirichlet boundary condition). The value of $C$ would need to be adjusted if `use_mobility = true` and/or if `use_relperm = true`.
```

Comment:
Last sentence should be (i think): The value of $C$ is simply $1/L$ if `use_mobility = true` and `use_relperm = true`.

Reply:
Ok